lib/sidekiq/cli.rb

```diff
@@ -231,6 +231,10 @@ def boot_system
           end
           require 'sidekiq/rails'
           require File.expand_path("#{options[:require]}/config/environment.rb")
+
+          if ::Rails::VERSION::MAJOR > 4 && ::Rails.env.development?
```

Comment:
Shouldn't this check if Rails.configuration.cache_classes is set to false instead of relying on a specific environment name?